app/routes/upload.step1.tsx

```diff
@@ -0,0 +1,130 @@
+import { ActionFunctionArgs } from '@remix-run/node'
+import { Form, redirect, useNavigation, useRouteError } from '@remix-run/react'
+import { useState } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+import { Button } from '~/components/ui/button'
+import { cn } from '~/lib/utils'
+import { supabase } from '~/supabaseClient'
+
+// Action to handle form submission
+export const action = async ({ request }: ActionFunctionArgs) => {
+  const body = await request.formData()
+  const file = body.get('cv-file') as File | null
+  if (!file) throw Error('Could not get file.')
+  if (
+    file &&
+    ![
+      'application/pdf',
+      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    ].includes(file.type)
+  )
+    throw Error('Only PDF and DOCX files are allowed.')
+
+  const uniqueFileName = `${uuidv4()}-${file.name}`
+  const { data, error } = await supabase.storage
+    .from('resumes')
+    .upload(`public/${uniqueFileName}`, file) // this is just for testing now. will be secured later
+
+  if (error) {
```

Comment:
**issue**: Consider expanding error handling to provide user-friendly messages instead of logging to the console. This will improve the user experience by providing immediate feedback on what went wrong.

Reply:
error.error which contains error message is thrown and ErrorBoundary cathes it to display the error message in UI